primal-bit/src/lib.rs

```diff
@@ -171,11 +170,21 @@ impl BitVec {
     /// }
     /// ```
     pub fn from_elem(nbits: usize, bit: bool) -> BitVec {
+        use std::ptr;
+
         let nblocks = nbits.checked_add(BITS - 1).expect("capacity overflow") / BITS;
+        let mut out_vec = Vec::with_capacity(nblocks);
+
+        unsafe {
+            out_vec.set_len(nblocks);
+            ptr::write_bytes(out_vec.as_mut_ptr(), if bit { !0 } else { 0 }, nblocks);
+        }
```

Comment:
The [`vec!` docs](https://doc.rust-lang.org/std/macro.vec.html) say that it essentially calls `.push(whatever.clone())` in a loop, I don't know how it could possibly do something as optimised as this without trait specialisation (which hasn't landed in stable as far as I know). It will probably get aggressively inlined and so forth but I can't imagine it turns into anything like this.